### DISCUSSION TITLE
About Network File System Volumes

### DISCUSSION THREAD

**Discussion by SalvaChiLlo** [minio/minio#16992] [Q&A] [answered]
Hi, I'm new to MinIO and I have some doubts about how can I deploy it. 
I've been asked to deploy a distributed multi node MinIO cluster.
It has to be deployed in a Kubernetes platform, which its storage backend relies on some Network File System, which is the only storage backend available currently.
As docs state **Network File System Volumes Break Consistency Guarantees** but my question is, what does this consistency mean? It's more focused on possible data corruption or just that data will eventually be consistent?

**Comment by Jazz9** [comment, marked as answer]
The issue is that the NFS layer will sometimes report that the file is written when it is really just in memory ( async nfs ). Power loss / connectivity issue etc may cause a issue with the file getting written and meanwhile Minio will have told the S3 client that the file is written.

If the NFS share is in SYNC mode - this shouldn't happen but its been fairly common for cheaper NFS NAS servers to not honor the sync flag.

**Reply by klauspost** [reply]
Yes. Also we see lockups and other errors occur under high load, so for the reason described above and for general stability we don't recommend it.
